GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -474,6 +477,51 @@ protected override void OnLoad(EventArgs e)
 
             // All app init is done, make all repo related similar to switching repos
             SetGitModule(this, new GitModuleEventArgs(new GitModule(Module.WorkingDir)));
+            bool isDashboard = _dashboard?.Visible ?? false;
+            ThreadHelper.FileAndForget(Task.Run(async () =>
+            {
+                if (isDashboard)
+                {
+                    PluginRegistry.InitializeGitHostersOnly();
+                    await this.SwitchToMainThreadAsync();
+                    if (_dashboard?.Visible ?? false)
+                    {
+                        // now that plugins are registered, populate Git-host-plugin actions on Dashboard, like "Clone GitHub repository"
+                        _dashboard.RefreshContent();
+                    }
+
+                    UpdateRepositoryHostsMenu();
+                }
+                else
+                {
+                    await InitializeAndRegisterAllPlugins();
+                    UpdateRepositoryHostsMenu();
+                }
+            }).ContinueWith(async _ =>
+            {
+                if (isDashboard)
+                {
+                    await InitializeAndRegisterAllPlugins();
+                }
+            }, TaskScheduler.Default));
```

Comment:
Why not
```suggestion
                if (isDashboard)
                {
                    await TaskScheduler.Default;
                    await InitializeAndRegisterAllPlugins();
                }
            });
```
without yet another `Task.Run` around?